File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/food-chain/food_chain.py
def recite(start_verse, end_verse):
    """Generate the lyrics of 'I Know an Old Lady Who Swallowed a Fly'."""
    
    animals = [
        ("fly", "I don't know why she swallowed the fly. Perhaps she'll die."),
        ("spider", "It wriggled and jiggled and tickled inside her."),
        ("bird", "How absurd to swallow a bird!"),
        ("cat", "Imagine that, to swallow a cat!"),
        ("dog", "What a hog, to swallow a dog!"),
        ("goat", "Just opened her throat and swallowed a goat!"),
        ("cow", "I don't know how she swallowed a cow!"),
        ("horse", "She's dead, of course!")
    ]
    
    verses = []
    
    for i in range(start_verse - 1, end_verse):
        current_animal, comment = animals[i]
        verse = [f"I know an old lady who swallowed a {current_animal}."]
        
        if i == len(animals) - 1:  # Special case for horse
            verses.extend(verse)
            verses.append(comment)
            continue
            
        verses.extend(verse)
        verses.append(comment)
        
        if i > 0:
            for j in range(i, 0, -1):
                predator = animals[j][0]
                prey = animals[j-1][0]
                if prey == "spider":
                    verses.append(f"She swallowed the {predator} to catch the {prey} that wriggled and jiggled and tickled inside her.")
                else:
                    verses.append(f"She swallowed the {predator} to catch the {prey}.")
            
            verses.append(animals[0][1])
        
        # Add empty line between verses except for the last one
        if i < end_verse - 1:
            verses.append("")
    
    return verses

File: src/2025-03-04-03-28-40--deepseek-chat-0304/java/exercises/practice/food-chain/src/main/java/FoodChain.java
class FoodChain {
    private static final String[] ANIMALS = {
        "fly", "spider", "bird", "cat", "dog", "goat", "cow", "horse"
    };
    
    private static final String[] DESCRIPTIONS = {
        "",
        "It wriggled and jiggled and tickled inside her.",
        "How absurd to swallow a bird!",
        "Imagine that, to swallow a cat!",
        "What a hog, to swallow a dog!",
        "Just opened her throat and swallowed a goat!",
        "I don't know how she swallowed a cow!",
        "She's dead, of course!"
    };
    
    private String buildChain(int verse) {
        StringBuilder chain = new StringBuilder();
        for (int i = verse - 1; i > 0; i--) {
            chain.append("She swallowed the " + ANIMALS[i] + " to catch the " + ANIMALS[i-1]);
            if (i == 2) {
                chain.append(" that wriggled and jiggled and tickled inside her");
            }
            chain.append(".\n");
        }
        return chain.toString();
    }

    String verse(int verse) {
        if (verse < 1 || verse > 8) {
            throw new IllegalArgumentException("Verse number must be between 1 and 8");
        }
        
        if (verse == 8) {
            return "I know an old lady who swallowed a horse.\nShe's dead, of course!\n";
        }
        
        StringBuilder sb = new StringBuilder();
        sb.append("I know an old lady who swallowed a " + ANIMALS[verse-1] + ".\n");
        if (verse > 1) {
            sb.append(DESCRIPTIONS[verse-1] + "\n");
        }
        sb.append(buildChain(verse));
        sb.append("I don't know why she swallowed the fly. Perhaps she'll die.\n");
        return sb.toString();
    }

    String verses(int startVerse, int endVerse) {
        if (startVerse < 1 || endVerse > 8 || startVerse > endVerse) {
            throw new IllegalArgumentException("Verses must be between 1 and 8, and startVerse must be <= endVerse");
        }
        
        StringBuilder sb = new StringBuilder();
        for (int i = startVerse; i <= endVerse; i++) {
            sb.append(verse(i));
            if (i < endVerse) {
                sb.append("\n");
            }
        }
        return sb.toString();
    }
}
